internal/store/postgres/asset_repository.go

```diff
@@ -26,20 +26,57 @@ type AssetRepository struct {
 
 // GetAll retrieves list of assets with filters
 func (r *AssetRepository) GetAll(ctx context.Context, flt asset.Filter) ([]asset.Asset, error) {
-	builder := r.getAssetSQL().Offset(uint64(flt.Offset))
-	size := flt.Size
+	fields := flt.IncludeFields
+
+	var builder sq.SelectBuilder
+	if len(fields) == 0 {
+		builder = r.getAssetSQL().Offset(uint64(flt.Offset))
```

Comment:
Are these changes necessary since search assets can also serve a similar purpose using ES?

Last time we came across changing these functionalities - we were aligning on enhancing the searchAssets API itself and consolidate both APIs into one. Context - https://github.com/raystack/compass/issues/201#issue-1556187908

@anjali9791 @sudo-suhas 

PS. search API also supports pagination now.

Reply:
Thanks for bringing this up @bsushmith. In my initial discussion with @anjali9791 I missed bringing it up.